src/sql/workbench/browser/modelComponents/declarativeTable.component.ts

```diff
@@ -205,6 +225,28 @@ export default class DeclarativeTableComponent extends ComponentBase implements
 
 	public setProperties(properties: { [key: string]: any; }): void {
 		super.setProperties(properties);
+		let dataChangedFlag: boolean = false;
+		if (this.columns) {
+			this.columns.forEach((cols, idx) => {
+				if (cols.showCheckAll) {
+					this.data.forEach((row) => {
+						if (row[idx] !== cols.isChecked) {
+							dataChangedFlag = true;
+						}
+						row[idx] = cols.isChecked;
+					});
+				}
+			});
+
+			//Creating a data change event if a cell value was changed due to check all value
+			this._changeRef.detectChanges();
+			if (dataChangedFlag) {
+				this.fireEvent({
+					eventType: ComponentEventType.onDidChange,
+					args: this.data
```

Comment:
We shouldn't be sending different args for the same event type - currently this event is fired with the cell changed data, not the entire set of data. So instead you should probably just fire this event in the forEach above for each specific cell that changed.

Reply:
Won't that fire many events for a single "check all" actions?

Reply:
Well, yes. But we can't just be sending random stuff in the event - how will the person listening on the other end know what to expect? All listeners would have to update their code to be able to handle either value being passed.

During your testing keep an eye out for perf issues though - I am slightly worried about sending hundreds of events at once if there's a ton of columns. But I think generally it's ok to do this for now and then as long as we don't see any immediate issues just keep an eye out for any customers hitting perf issues after this change.

Reply:
I'm testing on a table with 100 rows and it seems to be working fine. We can always change to firing one event later if there is big perf issue